nil/services/nilservice/service.go

```diff
@@ -213,25 +212,32 @@ func getRawApi(
 		panic("unsupported run mode for raw API")
 	}
 
-	shardApis := make(map[types.ShardId]rawapi.ShardApi)
+	nodeApiBuilder := rawapi.NodeApiBuilder()
 	for shardId := range types.ShardId(cfg.NShards) {
 		var err error
 		if slices.Contains(myShards, uint(shardId)) {
-			shardApis[shardId] = rawapi.NewLocalShardApi(shardId, database, txnPools[shardId], readonly, cfg.EnableDevApi)
-			if assert.Enable {
-				api, ok := shardApis[shardId].(*rawapi.LocalShardApi)
-				check.PanicIfNot(ok)
-				shardApis[shardId], err = rawapi.NewLocalRawApiAccessor(shardId, api)
+			err = nodeApiBuilder.WithLocalShardApiRo(shardId, database, txnPools[shardId], cfg.EnableDevApi)
```

Comment:
`ShardApiRo` shouldn't accept txPools. Let's remove this argument or pass `nil` explicitly.